flocker/provision/_install.py

```diff
@@ -268,23 +379,32 @@ def task_install_flocker(
         distribution=None,
         package_source=PackageSource()):
     """
-    Install flocker on a distribution.
+    Install flocker cluster on a distribution.
+
+    The ClusterHQ repo is added for downloading latest releases.  If
+    ``package_source`` contains a branch, then a BuildBot repo will also
+    be added to the package search path, to use in-development packages.
+    Note, the ClusterHQ repo is always enabled, to provide dependencies.
 
     :param bytes distribution: The distribution the node is running.
     :param PackageSource package_source: The source from which to install the
         package.
     """
     if package_source.branch:
+        # A development branch has been selected - add its Buildbot repo
+        use_development_branch = True
         result_path = posixpath.join(
             '/results/omnibus/', package_source.branch, distribution)
         base_url = urljoin(package_source.build_server, result_path)
     else:
-        base_url = None
+        use_development_branch = False
+
     if distribution == 'ubuntu-14.04':
         commands = [
-            # Ensure add-apt-repository command is available
+            # Ensure add-apt-repository command and HTTPS URLs are supported
```

Comment:
Will Buildbot tests ail if either of these are not done? Under which circumstances does this need to be done? (I vaguely remember this being a solution to a problem with a stripped down instance for a Docker image).